src/nanotron/config/config.py

```diff
@@ -375,10 +336,30 @@ def as_dict(self) -> dict:
         return serialize(self)
 
 
-def get_config_from_dict(args: dict, config_class: Type[Config] = Config):
+def get_config_from_dict(
+    config_dict: dict, config_class: Type = Config, skip_unused_config_keys: bool = True, skip_null_keys: bool = True
+):
+    """Get a config object from a dictionary
+
+    Args:
+        args: dictionary of arguments
+        config_class: type of the config object to get as a ConfigTypes (Config, LightevalConfig, LightevalSlurm) or str
+        skip_unused_config_keys: whether to skip unused first-level keys in the config file (for config with additional sections)
+        skip_null_keys: whether to skip keys with value None at first and second level
+    """
+    if skip_unused_config_keys:
+        config_dict = {
+            field.name: config_dict[field.name] for field in fields(config_class) if field.name in config_dict
+        }
+    if skip_null_keys:
```

Comment:
should we add warnings when we skip keys or no need to?

Reply:
yes sounds good